invader_payment_paypal/services/payment_paypal.py

```diff
@@ -0,0 +1,232 @@
+                        "value": transaction.amount,
+                    }
+                }
+            ],
+            "application_context": {
+                "landing_page": "LOGIN",
+                "user_action": "PAY_NOW",
+                "return_url": return_url,
+                "cancel_url": cancel_url,
+            },
+        }
+
+    def checkout_order(
+        self, target, payment_mode_id, return_url, cancel_url, **params
+    ):
+        transaction_obj = self.env["payment.transaction"]
+        payable = self.payment_service._invader_find_payable_from_target(
+            target, **params
+        )
+
+        payment_mode = self.env["account.payment.mode"].browse(payment_mode_id)
+        self.payment_service._check_provider(payment_mode, "paypal")
+
+        transaction = transaction_obj.create(
+            payable._invader_prepare_payment_transaction_data(payment_mode)
+        )
+        payable._invader_set_payment_mode(payment_mode)
+        data = self._prepare_paypal_order(
+            transaction, return_url, cancel_url, **params
+        )
+        paypal_order = self._create_paypal_order(transaction.acquirer_id, data)
+        transaction.write({"acquirer_reference": paypal_order["id"]})
+        for link in paypal_order["links"]:
+            if link["rel"] == "approve":
+                return {"redirect_to": link["href"]}
+        # In case of success paypal always provide the approve url
+        # Passing in the following code should never happen
+        _logger.error("Invalid Paypal response {}".format(paypal_order))
+        raise UserError(_("Fail to create the payment"))
+
+    def _validator_normal_return(self):
+        schema = {
+            "token": {"type": "string"},
+            "success_redirect": {"type": "string"},
+            "cancel_redirect": {"type": "string", "required": True},
+        }
+        schema.update(self.payment_service._invader_get_target_validator())
+        return Validator(schema, allow_unknown=True)
+
+    def _validator_return_normal_return(self):
+        schema = {"redirect_to": {"type": "string"}}
+        return Validator(schema, allow_unknown=True)
+
+    def _capture_transaction(self, transaction, raise_error=False):
+        acquirer = transaction.acquirer_id
+        token = self._get_api_token(acquirer)
+        response = requests.post(
+            self._get_api_url(
+                acquirer,
+                "v2/checkout/orders/{}/capture".format(
+                    transaction.acquirer_reference
+                ),
+            ),
+            headers={
+                "Content-Type": "application/json",
+                "Authorization": token,
+            },
+        )
+        if response.status_code != 201:
+            try:
+                self._catch_paypal_error(response)
+            except UserError as e:
+                if raise_error:
+                    raise
+                else:
```

Comment:
@sebastienbeau We must keep a trace of the the initial error. Otherwise wil lnot be possible to understand what's wrong...